controller/tests/test_waln.py

```diff
@@ -0,0 +1,58 @@
+import pytest
+from unittest.mock import patch
+from controller.src.board import wlan, credentials
+
+CYW43_STATUS_CODES = {
+    "CYW43_LINK_BADAUTH": -3,
+    "CYW43_LINK_NONET": -2, # No matching SSID found (out of range or down) 
+    "CYW43_LINK_FAIL": -1,
+    "CYW43_LINK_DOWN": 0,
+    "CYW43_LINK_JOIN": 1,
+    "CYW43_LINK_NOIP": 2,
+    "CYW43_LINK_UP": 3
+}
+
+test_credentials = credentials.WLANCredentials("id", "pass")
+
+def test_connect_success():
+    wlan._wlan.connect_state = False
+    wlan._wlan.status_code = CYW43_STATUS_CODES["CYW43_LINK_UP"]
+    assert test_credentials == wlan.connect(test_credentials)
+
+def test_connect_failure():
+    for key in ["CYW43_LINK_FAIL", "CYW43_LINK_NONET", "CYW43_LINK_BADAUTH"]:
+        wlan._wlan.status_code = CYW43_STATUS_CODES[key]
+        with pytest.raises(wlan.WLANConnectionFailure):
+            wlan.connect(test_credentials)
+
+@patch('time.sleep', return_value=None)
+def test_connect_timeout(patched_time_sleep):
+    for key in ["CYW43_LINK_DOWN", "CYW43_LINK_JOIN", "CYW43_LINK_NOIP"]:
+        wlan._wlan.status_code = CYW43_STATUS_CODES[key]
+        with pytest.raises(wlan.WLANConnectionTimeout):
+            wlan.connect(test_credentials)
+
+def test_connect_bad_arg():
+    with pytest.raises(TypeError):
+        wlan.connect("these are not credentials")
+
+def test_connect_already_connected():
+    wlan._wlan.connect_state = True
+    with pytest.raises(wlan.WLANConnectionActiveError):
+        wlan.connect(test_credentials)
+
+def test_disconnect():
+    wlan._wlan.connect_state = True
+    wlan.disconnect()
+    assert wlan._wlan.connect_state == False
+
+def test_getCredentials():
```

Comment:
aaron@<computer> ~/git/luminosity-led (LL-117)
$ pytest controller/tests/test_waln.py 
====================================================================== test session starts =======================================================================
platform win32 -- Python 3.10.7, pytest-7.2.0, pluggy-1.0.0
rootdir: C:\Users\aaron\git\luminosity-led
plugins: mock-3.10.0
collected 9 items

controller\tests\test_waln.py ......F..                                                                                                                     [100%]

============================================================================ FAILURES ============================================================================ 
______________________________________________________________________ test_getCredentials _______________________________________________________________________

    def test_getCredentials():
>       assert isinstance(wlan.getCredentials(), credentials.WLANCredentials)
E       AssertionError: assert False
E        +  where False = isinstance(None, <class 'controller.src.board.credentials.WLANCredentials'>)
E        +    where None = <function getCredentials at 0x000001C9C5CF7370>()
E        +      where <function getCredentials at 0x000001C9C5CF7370> = wlan.getCredentials
E        +    and   <class 'controller.src.board.credentials.WLANCredentials'> = credentials.WLANCredentials

controller\tests\test_waln.py:50: AssertionError
==================================================================== short test summary info ===================================================================== 
FAILED controller/tests/test_waln.py::test_getCredentials - AssertionError: assert False
================================================================== 1 failed, 8 passed in 0.94s =================================================================== 
(.venv)

Reply:
Thanks for pointing this out, it should be a quick fix.